dashboard/lib/projects_list.rb

```diff
@@ -72,6 +72,37 @@ def fetch_published_projects(project_group, limit:, published_before:)
       fetch_published_project_types([project_group.to_s], limit: limit, published_before: published_before)
     end
 
+    def fetch_featured_published_projects
+      featured_published_projects = {}
+      PUBLISHED_PROJECT_TYPE_GROUPS.keys.each do |project_type|
+        featured_published_projects[project_type] = fetch_featured_projects_by_type(project_type)
+      end
+      featured_published_projects
+    end
+
+    def project_and_featured_project_fields
```

Comment:
This is now a duplicate of code in the projects_controller.  Either I should move it to a shared place, if so, where? Or these fields should be different. I'll need to get student info for the owner of the project before I can add these projects to the list to be rendered in the gallery.  Does that mean I need a 3 way join? Or should I refactor to re-use that join on the users table that already exists?

Reply:
Yes, I think you should use a 3-way join here. One gotcha is that the syntax inside the `join` can be tricky, because `id` can be ambiguous, since that field exists in multiple tables. Here are the docs for how to specify which table you are referring to when you specify an `id` in 3-way joins, as well which table is used by default if you do not specify: https://github.com/jeremyevans/sequel/blob/master/doc/querying.rdoc#join-conditions

Whether to extract and share or just leave separate is a judgement call, and either could be ok. In this case, it seems that the Sequel queries are complex enough that it would be simplest not to extract common parts of them. Similarly, I would recommend not extracting common sub-lists from these lists of fields, either. 

If you do decide to extract common code between here and projects_controller, then it seems like this file would be the right place to extract the common code into. If you don't like having stuff split up across files, you could even move `combine_projects_and_featured_projects_data` and everything it calls into this file.